src/shared/tpm2-util.c

```diff
@@ -4194,7 +4194,9 @@ Tpm2Support tpm2_support(void) {
                 support |= TPM2_SUPPORT_FIRMWARE;
 
 #if HAVE_TPM2
-        support |= TPM2_SUPPORT_SYSTEM;
+        r = dlopen_tpm2();
+        if (r >= 0)
+                support |= TPM2_SUPPORT_SYSTEM;
```

Comment:
hmm, we could also add another flag for that. i.e. might be interesting to distinguish "systemd would support it if you installed some more .so" and "systemd would not support it, fuck you"...

i.e. add TPM2_SUPPORT_LIBRARIES or so